src/Init/Data/Int/DivMod/Lemmas.lean

```diff
@@ -2354,6 +2354,76 @@ theorem bmod_neg_bmod : bmod (-(bmod x n)) n = bmod (-x) n := by
   apply (bmod_add_cancel_right x).mp
   rw [Int.add_left_neg, ← add_bmod_bmod, Int.add_left_neg]
 
+/- ### ediv -/
+
+theorem ediv_lt_self_of_pos_of_one_lt (x y : Int) (hx' : 0 < x) (hy' : 1 < y) :
+    x / y < x := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
+  apply Nat.div_lt_self (by omega) (by omega)
+
+-- note that x < y → x / y = 0
+theorem nonneg_ediv_of_pos_of_nonneg (x y : Int) (hx' : 0 < x) (hy' : 0 ≤ y) :
+    0 ≤ x / y := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
```

Comment:
```suggestion
theorem ediv_nonneg_of_nonneg_of_nonneg (x y : Int) (hx' : 0 ≤ x) (hy' : 0 ≤ y) :
    0 ≤ x / y := by
  rw [Int.div_def]
  unfold Int.ediv
  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
  exact Int.ofNat_zero_le (xn / yn)
```

If you look higher up in the file, the naming convention is `ediv_nonneg` instead of `nonneg_ediv`. Also, you can weaken `hx'`.

Reply:
updated the name